File: src/main/resources/static/script/type.js
import Pair from "/script/pair.js";

export default class Type {
    typeEnum;
    typeName;

    constructor(typeEnum, typeName) {
        this.typeEnum = typeEnum;
        this.typeName = typeName;
    }

    getTypeEnum() {
        return this.typeEnum;
    }

    getTypeName() {
        return this.typeName;
    }
}

export const TYPES = new Array(
    new Type("Empty", ""),
    new Type("Normal", "ノーマル"), 
    new Type("Heating", "ほのお"),
    new Type("Water", "みず"),
    new Type("Voltage", "でんき"),
    new Type("Leaf", "くさ"),
    new Type("Ice", "こおり"),
    new Type("Combat", "かくとう"),
    new Type("Poison", "どく"),
    new Type("Temblor", "じめん"),
    new Type("Aviation", "ひこう"),
    new Type("Esper", "エスパー"),
    new Type("Bug", "むし"),
    new Type("Rock", "いわ"),
    new Type("Ghost", "ゴースト"),
    new Type("Dragon", "ドラゴン"),
    new Type("Mischief", "あく"),
    new Type("Steel", "はがね"),
    new Type("Fairy", "フェアリー")
);

export const RESULT_SYMBOLS = new Array(
    new Pair(-3, "×"), 
    new Pair(-2, "▲"), 
    new Pair(-1, "△"), 
    new Pair(1, "○"), 
    new Pair(2, "◎")
);

/**
 * * タイプ名取得
 */
export function getTypeName(typeEnum) {
    for(let i = 0; i < TYPES.length; i++) {
        if(typeEnum == TYPES[i].getTypeEnum()) return TYPES[i].getTypeName();
    }

    return "";
}

/**
 * * 結果値から表示記号取得
 */
export function getResultSymbol(result) {
    for(let i = 0; i < RESULT_SYMBOLS.length; i++) {
        if(result == RESULT_SYMBOLS[i].getKey()) return RESULT_SYMBOLS[i].getValue();
    }

    return "";
}

File: src/main/java/com/mitsurin/tools/creating_best_party/model/compatibility/Type.java
package com.mitsurin.tools.creating_best_party.model.compatibility;

public enum Type {
    Empty(""),
    Normal("ノーマル"),
    Heating("ほのお"),
    Water("みず"),
    Voltage("でんき"),
    Leaf("くさ"),
    Ice("こおり"),
    Combat("かくとう"),
    Poison("どく"),
    Temblor("じめん"),
    Aviation("ひこう"),
    Esper("エスパー"),
    Bug("むし"),
    Rock("いわ"),
    Ghost("ゴースト"),
    Dragon("ドラゴン"),
    Mischief("あく"),
    Steel("はがね"),
    Fairy("フェアリー");

    public final String name;

    Type(String name) {
        this.name = name;
    }

    public static Type getTypeByTypeName(String name) {
        for (Type type : Type.values()) {
            if(name.equals(type.toString())) return type;
        }
        return null;
    }

    public static Type getTypeByName(String name) {
        for (Type type : Type.values()) {
            if(name.equals(type.name)) return type;
        }
        return null;
    }
}
